lib/install/phases/pull.go

```diff
@@ -107,11 +107,21 @@ type pullExecutor struct {
 
 // Execute executes the pull phase
 func (p *pullExecutor) Execute(ctx context.Context) error {
-	err := p.pullUserApplication()
-	if err != nil {
-		return trace.Wrap(err)
+	// If the list of packages to pull was explicitly provided, pull only those
+	// (e.g. during join), otherwise pull the entire user application (e.g.
+	// during initial installation).
```

Comment:
When is the application package required during the installation? Can you elaborate here?
Also the `pullUserApplication` hints at the possibility of skipping the app pull on regular nodes - is it something this PR can also do?
And if the pulling ends up requiring the application package only on a single node - how about splitting the pull step into pull app and pull packages which will make the decision more emphasized as now it is pretty much buried inside and not explicit about when and whether the application package is pulled.